src/extensions/scratch3_microbit/index.js

```diff
@@ -317,7 +327,11 @@ class MicroBit {
 
         // cancel disconnect timeout and start a new one
         window.clearInterval(this._timeoutID);
-        this._timeoutID = window.setInterval(this.disconnect, BLETimeout);
+        this._timeoutID = window.setInterval(
+            // send an error to the BLE socket
+            this._ble._sendDisconnectError.bind(this._ble, BLEDataStoppedError),
```

Comment:
per discussion we decided to make this a public function and rename to `triggerDisconnectError` (or similar)

Reply:
Line 302 (in `_onConnect`) also needs the change to call this function on `_ble` instead of just `disconnect`

Reply:
I have chosen to name it `handleDisconnectError` and will also update two other error handlers in both BT/BLE to also use "handle" in their name (as opposed to "send")